conjure-go-client/httpclient/metrics_test.go

```diff
@@ -300,3 +300,41 @@ func TestMetricsMiddleware_FailedTLSHandshake(t *testing.T) {
 	assert.False(t, success, "successful tls handshake attempt registered")
 	assert.True(t, failure, "no failed tls handshake attempt registered")
 }
+
+func TestMetricsMiddleware_InFlightRequests(t *testing.T) {
+	rootRegistry := metrics.NewRootMetricsRegistry()
+	ctx := metrics.WithRegistry(context.Background(), rootRegistry)
+	serviceNameTag := metrics.MustNewTag("service-name", "test-service")
+
+	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		dialerMetric := rootRegistry.Counter(httpclient.MetricInFlightConns, serviceNameTag).Count()
+		assert.Equal(t, int64(1), dialerMetric, "%s should be nonzero during a request", httpclient.MetricInFlightConns)
+		clientMetric := rootRegistry.Counter(httpclient.MetricInFlightRequests, serviceNameTag).Count()
+		assert.Equal(t, int64(1), clientMetric, "%s should be nonzero during a request", httpclient.MetricInFlightRequests)
+		w.WriteHeader(200)
+	}))
+	defer srv.Close()
+
+	client, err := httpclient.NewClient(
+		httpclient.WithBaseURLs([]string{srv.URL}),
+		httpclient.WithServiceName("test-service"),
+		httpclient.WithMetrics())
+	httpclient.WithIdleConnTimeout(time.Second)
+	require.NoError(t, err)
+
+	_, err = client.Get(ctx)
+	require.NoError(t, err)
+
+	// do the request a second time to assert the connection is reused
+	_, err = client.Get(ctx)
+	require.NoError(t, err)
+
+	clientMetric := rootRegistry.Counter(httpclient.MetricInFlightRequests, serviceNameTag).Count()
+	assert.Equal(t, int64(0), clientMetric, "%s should be zero after a request", httpclient.MetricInFlightRequests)
+
+	dialerMetric := rootRegistry.Counter(httpclient.MetricInFlightConns, serviceNameTag)
+	assert.Equal(t, int64(1), dialerMetric.Count(), "%s should be nonzero immediately after a request", httpclient.MetricInFlightConns)
+	srv.Close()
+	time.Sleep(time.Second)
```

Comment:
were you hitting races without this? From the docs `srv.Close()` will block until all requests are complete, so it surprises me that you would need this.

Reply:
I thought I was but there was a lot of iteration going on. I removed it but added a comment for future-me to not be frustrated by a flaky test 😅